tests/test_example_usage.py

```diff
@@ -0,0 +1,351 @@
+import os
+import sys
+import time
+
+import pexpect
+import pytest
+
+
+class CliTester:
+    DEFAULT_START_WAIT_TIME = 0.2
+    DEFAULT_PROCESS_TIMEOUT = 1
+    title = None
+    footer = None
+    description = None
+
+    def __init__(self, cfg_dir):
+        os.environ["NM_CONFIG_DIR"] = str(cfg_dir)
+        os.environ["COVERAGE_PROCESS_START"] = ".coveragerc"
+        self._step_results = []
+        self._async_procs = []
+
+    @staticmethod
+    def _cmd_to_coverage(cmd):
+        cmd = cmd.replace("inet-nm-", "inet_nm.cli_")
+        cmd = cmd.replace("-", "_")
+        return f"coverage run -m {cmd}"
+
+    def run_step(
+        self,
+        cmd,
+        args=None,
```

Comment:
Why not default to `[]`? Saves the `.. or []` down the line?

Reply:
[It goes against the python recommendations](https://docs.python.org/3/reference/compound_stmts.html#function-definitions)

> Default parameter values are evaluated from left to right when the function definition is executed. This means that the expression is evaluated once, when the function is defined, and that the same “pre-computed” value is used for each call. This is especially important to understand when a default parameter value is a mutable object, such as a list or a dictionary: if the function modifies the object (e.g. by appending an item to a list), the default parameter value is in effect modified. This is generally not what was intended. A way around this is to use None as the default, and explicitly test for it in the body of the function, e.g.:
```py
def whats_on_the_telly(penguin=None):
    if penguin is None:
        penguin = []
    penguin.append("property of the zoo")
    return penguin
```

Reply:
🤯 wow! I Never knew! Makes me wonder how much of my Code only works by accident...